app/src/main/java/io/supertokens/session/SuperTokensHttpURLConnection.java

```diff
@@ -39,6 +40,21 @@ public static HttpURLConnection newRequest(URL url, PreConnectCallback preConnec
         if ( applicationContext == null ) {
             throw new IllegalAccessException("Context is null");
         }
+
+        boolean doNotDoInterception = !Utils.shouldDoInterceptionBasedOnUrl(url.toString(), SuperTokens.config.apiDomain, SuperTokens.config.cookieDomain);
+
+        if (doNotDoInterception) {
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+
+            // TODO NEMI: Replace this with pre api hook when implemented. This also means we need to have a non superotkens pre API hook action for HTTPURLConnection
+            if (preConnectCallback != null) {
+                preConnectCallback.doAction(connection);
+            }
```

Comment:
No.. we don't call pre API hook in this case. It's not happening in website repo right?

Reply:
So for httpurl connection we only accept URL as the input (because accepting a connection object makes it more complicated to set values and retry requests). If we assume that users may call this function for requests that they dont want to use SuperTokens for then the only way for them to add request headers, method etc would be by the pre connect callback

Reply:
This is different in supertokens website because fetch lets you pass a config too, but http url connection only lets you set things on the connection object